dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/arbitrum/cow_protocol_arbitrum_batches.sql

```diff
@@ -15,35 +15,44 @@
     )
 }}
 
-WITH
+with
 batch_counts as (
     select try_cast(date_trunc('day', s.evt_block_time) as date) as block_date,
            s.evt_block_number,
            s.evt_block_time,
            s.evt_tx_hash,
-           solver,
-           name,
+           solver_info.solver,
+           solver_info.name,
            sum(
                case
                    when selector != 0x2e1a7d4d -- unwrap
                     and selector != 0x095ea7b3 -- approval
                        then 1
                    else 0
-                end)                                                as dex_swaps,
+               end
+           ) as dex_swaps,
            sum(case when selector = 0x2e1a7d4d then 1 else 0 end) as unwraps,
            sum(case when selector = 0x095ea7b3 then 1 else 0 end) as token_approvals
     from {{ source('gnosis_protocol_v2_arbitrum', 'GPv2Settlement_evt_Settlement') }} s
         left outer join {{ source('gnosis_protocol_v2_arbitrum', 'GPv2Settlement_evt_Interaction') }} i
             on i.evt_tx_hash = s.evt_tx_hash
             {% if is_incremental() %}
-            AND {{ incremental_predicate('i.evt_block_time') }}
+            and {{ incremental_predicate('i.evt_block_time') }}
             {% endif %}
-        join {{ ref('cow_protocol_arbitrum_solvers') }}
-            on solver = address
+        join lateral (
+            select
+                t."from" as solver,
+                solvers.name as name
+            from {{ source('arbitrum', 'traces') }} t
+                inner join {{ ref('cow_protocol_arbitrum_solvers') }} solvers on t."from" = solvers.address
+            where t.tx_hash = s.evt_tx_hash
```

Comment:
```suggestion
            where t.tx_hash = s.evt_tx_hash
                and t.block_date = date_trunc('day', s.evt_block_time)
                and t.block_number = s.evt_block_number
```

Reply:
is it possible to include these fields in join conditions, to help performance?
not sure if the decoded table has `evt_block_date`, hence the conversion, but either would work

Reply:
just pushed a change with your suggestion.